tests/test_bookmarks.py

```diff
@@ -19,6 +19,37 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.start_date = '2021-04-01T00:00:00Z'
 
+    def max_bookmarks_by_stream(self, sync_records):
+        """
+        Return the maximum value for the replication key for each stream
+        which is the bookmark expected value.
+
+        Comparisons are based on the class of the bookmark value. Dates will be
+        string compared which works for ISO date-time strings
+        """
+        max_bookmarks = {}
+        for stream, batch in sync_records.items():
+            if stream != 'metafields':
+                upsert_messages = [m for m in batch.get('messages') if m['action'] == 'upsert']
+            else:
+                upsert_messages = [m for m in batch.get('messages') if m['action'] == 'upsert' and m.get('data').get('owner_resource') == 'shop']
```

Comment:
Add a comment here regarding the specific behavior of metafields stream.

Reply:
added comments